src/lib/core/selection/selection.ts

```diff
@@ -0,0 +1,168 @@
+import {Subject} from 'rxjs/Subject';
+
+
+/**
+ * Class to be used to power selecting one or more options from a list.
+ * @docs-private
+ */
+export class MdSelectionModel {
+  constructor(
+    private _options: any[],
```

Comment:
I think we can avoid keeping `_options` in this model completely if we add it as an argument to `selectAll`. That would make this class much simpler (no update logic, remove `_verifyExistence`)